public/pages/workflow_builder/components/input_field_list.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { IComponentField } from '../../../component_types';
+import { TextField, JsonField, SelectField } from './input_fields';
+
+/**
+ * A helper component to format all of the input fields for a component. Dynamically
+ * render based on the input type.
+ */
+
+interface InputFieldListProps {
+  inputFields?: IComponentField[];
+}
+
+export function InputFieldList(props: InputFieldListProps) {
+  return (
+    <EuiFlexItem>
+      {props.inputFields?.map((field, idx) => {
+        let el;
+        switch (field.type) {
```

Comment:
nit: instead of using a switch statement, consider refactoring it to use a mapping object for better clarity and scalability. This also makes adding new input types in the future easier

Reply:
Thanks. Likely this will be moved into more complex helper fns as the number of object types will grow, and the complexity to render them will grow. Prefer to just leave as a simple switch for now.